diff --git a/dist/alphaTab.core.mjs b/dist/alphaTab.core.mjs
index b6203da6066c95efcf66298fe48a690f144597a5..6e6dfafee46a9679adc001c9ea65ebfd5c44a91e 100644
--- a/dist/alphaTab.core.mjs
+++ b/dist/alphaTab.core.mjs
@@ -52712,7 +52712,9 @@ class EffectBandSlot {
 	canBeUsed(band) {
 		if (!(!this.shared.uniqueEffectId && band.info.canShareBand || band.info.effectId === this.shared.uniqueEffectId)) return false;
 		if (!this.shared.firstBeat) return true;
-		if (this.shared.lastBeat === band.firstBeat) return true;
+		if (this.shared.lastBeat === band.firstBeat) {
+			return this.bands.some((existingBand) => existingBand.info.effectId === band.info.effectId && existingBand.lastBeat === band.firstBeat);
+		}
 		if (this.shared.lastBeat.isBefore(band.firstBeat)) return true;
 		if (this.shared.lastBeat.isBefore(this.shared.firstBeat)) return true;
 		return false;
diff --git a/dist/alphaTab.js b/dist/alphaTab.js
index 6da399cf7c1bea0beee994643890d4c9e75bde2e..6dc85c0db2942b013da14b003de9446d5fa3e739 100644
--- a/dist/alphaTab.js
+++ b/dist/alphaTab.js
@@ -52716,7 +52716,9 @@
 		canBeUsed(band) {
 			if (!(!this.shared.uniqueEffectId && band.info.canShareBand || band.info.effectId === this.shared.uniqueEffectId)) return false;
 			if (!this.shared.firstBeat) return true;
-			if (this.shared.lastBeat === band.firstBeat) return true;
+			if (this.shared.lastBeat === band.firstBeat) {
+				return this.bands.some((existingBand) => existingBand.info.effectId === band.info.effectId && existingBand.lastBeat === band.firstBeat);
+			}
 			if (this.shared.lastBeat.isBefore(band.firstBeat)) return true;
 			if (this.shared.lastBeat.isBefore(this.shared.firstBeat)) return true;
 			return false;
